### PR TITLE
Support workflow name as first argument of HOOK_WORFLOW. Previously a…

### PR DESCRIPTION
**Pull request by arielalmendral**
… workflow filename was expected

### CODE REVIEW

File: devel/libenkf/include/ert/enkf/hook_manager.h [hunk 1]
```diff
@@ -29,27 +29,30 @@ extern "C" {
...
 
-  void                  hook_manager_init( hook_manager_type * hook_manager , const config_content_type * config);
-  void                  hook_manager_add_config_items( config_parser_type * config );
+  runpath_list_type          * hook_manager_get_runpath_list( hook_manager_type * hook_manager );
+  void                         hook_manager_export_runpath_list( const hook_manager_type * hook_manager );
+  void                         hook_manager_set_runpath_list_file( hook_manager_type * hook_manager , const char * path, const char * filename);
+  const char                 * hook_manager_get_runpath_list_file(const hook_manager_type * hook_manager);
+  void                         hook_manager_run_workflows( const hook_manager_type * hook_manager , hook_run_mode_enum run_mode , void * self);
+  const hook_workflow_type   * hook_manager_iget_hook_workflow(const hook_manager_type * hook_manager, int index);
+  int                          hook_manager_get_size(const hook_manager_type * hook_manager);
```
Comment (joakim-hove) [inline devel/libenkf/include/ert/enkf/hook_manager.h]: Please avoid these white-space changes; it makes review unnecessary difficult.

File: devel/libenkf/src/hook_manager.c [hunk 2]
```diff
@@ -82,37 +82,36 @@ runpath_list_type * hook_manager_get_runpath_list( hook_manager_type * hook_mana
...
-      vector_append_owned_ref(hook_manager->hook_workflow_list, hook , hook_workflow_free__);
-    }
+static void hook_manager_add_workflow( hook_manager_type * hook_manager , const char * workflow_name , hook_run_mode_enum run_mode) {
+  if (ert_workflow_list_has_workflow( hook_manager->workflow_list , workflow_name) ){
+    workflow_type * workflow = ert_workflow_list_get_workflow( hook_manager->workflow_list , workflow_name);
+    hook_workflow_type * hook = hook_workflow_alloc( workflow , run_mode );
+    vector_append_owned_ref(hook_manager->hook_workflow_list, hook , hook_workflow_free__);
+  }
+  else {
+    printf("WARNING: While hooking workflow: %s not recognized among the list of loaded workflows.", workflow_name);
```
Comment (joakim-hove) [inline devel/libenkf/src/hook_manager.c]: Warning should go on `stderr`.

File: devel/libenkf/src/hook_manager.c [hunk 3]
```diff
@@ -131,14 +130,14 @@ void hook_manager_add_config_items( config_parser_type * config ) {
 
     item = config_add_schema_item( config , QC_WORKFLOW_KEY , false );
     config_schema_item_set_argc_minmax(item , 1 , 1 );
-    config_schema_item_iset_type( item , 0 , CONFIG_EXISTING_PATH );
```
Comment (joakim-hove) [inline devel/libenkf/src/hook_manager.c]: The `QC_WORKFLOW` config key should continue to work as it always has.

File: devel/libenkf/src/hook_manager.c [hunk 4]
```diff
@@ -82,37 +82,36 @@ runpath_list_type * hook_manager_get_runpath_list( hook_manager_type * hook_mana
...
 
 
 
 void hook_manager_init( hook_manager_type * hook_manager , const config_content_type * config_content) {
 
-  /* Old stuff explicitly prefixed with QC - the  */
+  /* Old stuff explicitly prefixed with QC  */
   {
     if (config_content_has_item( config_content , QC_WORKFLOW_KEY)) {
-      const char * file_name = config_content_get_value_as_path(config_content , QC_WORKFLOW_KEY);
```
Comment (joakim-hove) [inline devel/libenkf/src/hook_manager.c]: The `QC_WORKFLOW` config key should continue to work as it always has.

File: devel/python/python/ert/enkf/hook_manager.py [hunk 5]
```diff
@@ -2,13 +2,25 @@
 import sys
 from ert.cwrap import BaseCClass, CWrapper
 from ert.enkf import ENKF_LIB
+from pandas.core import indexing
```
Comment (joakim-hove) [inline devel/python/python/ert/enkf/hook_manager.py]: Ehhh - what is this?

File: devel/libenkf/include/ert/enkf/hook_manager.h (hunk 1 above)
Comment (arielalmendral) [inline devel/libenkf/include/ert/enkf/hook_manager.h]: Understandable :)

File: devel/libenkf/src/hook_manager.c (hunk 2 above)
Comment (arielalmendral) [inline devel/libenkf/src/hook_manager.c]: Will do.

File: devel/libenkf/src/hook_manager.c (hunk 3 above)
Comment (arielalmendral) [inline devel/libenkf/src/hook_manager.c]: I thought you wanted a change in behavior for QC_WORKFLOW as well. Will keep old variant with path for compatibility.

File: devel/python/python/ert/enkf/hook_manager.py (hunk 5 above)
Comment (arielalmendral) [inline devel/python/python/ert/enkf/hook_manager.py]: Didn't see this, sorry for the noise.

**Comment by joakim-hove** [comment]
Good ;-)

**Pull request merged**